### PR TITLE
Lab7done

### PR DESCRIPTION
**Pull request by Jerry9925-csd**

### CODE REVIEW

File: src/test/java/lab7/MainTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    @Test
+    void testSearchRecipes_CaseInsensitive() {
+        DataService mockService = () -> List.of(
+                new Recipe(1, "CHICKEN Curry", "Spicy and tasty", "", 2, 10, 15, 25)
+        );
+
+        List<Recipe> result = Main.searchRecipes("chicken", mockService);
+        assertEquals(1, result.size());
+        assertEquals("CHICKEN Curry", result.get(0).name());
+    }
```
Comment (rmartin-sc) [inline src/test/java/lab7/MainTest.java:101]: You should include a test that exceptions thrown by the DataService are handled gracefully by returning an empty list
